nixos/tests/vscode-remote-ssh.nix

```
@@ -0,0 +1,145 @@
+import ./make-test-python.nix ({ lib, ... }@args: let
+  pkgs = args.pkgs.extend (self: super: {
+    stdenv = super.stdenv.override {
+      config = super.config // {
+        allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
+          "vscode" "vscode-with-extensions" "vscode-extension-ms-vscode-remote-remote-ssh"
+        ];
+      };
+    };
+  });
+
+  inherit (import ./ssh-keys.nix pkgs) snakeOilPrivateKey snakeOilPublicKey;
+
+  # The commit hash for stable VS Code in Nixpkgs
+  rev = "ee2b180d582a7f601fa6ecfdad8d9fd269ab1884";
+  shortRev = builtins.substring 0 8 rev;
+
+  vscode-server = pkgs.srcOnly {
```

```suggestion
  vscode-server = pkgs.srcOnly {
```
I'm trying to understand what you are doing but I'm failing.
I think you are mixing packages and tests.
Update script changing a test is fishy. A test should be static.
Also add comments to make obvious what you are doing.

---

This test includes the VS Code server package because on a normal machine the Remote SSH extension would download it but our tests run without networking so we need to download it and install it ourselves as part of the test.

I've added this as a comment as well